modules/element-manager/ajax.php

```diff
@@ -68,6 +79,50 @@ public function ajax_get_admin_page_data() {
 		wp_send_json_success( $data );
 	}
 
+	private function get_element_manager_permissions_promotion(): array {
+		$promotion_data = [
+			'pro' => [
+				'text' => esc_html__( 'Upgrade Now', 'elementor' ),
+				'url' => self::FREE_TO_PRO_PERMISSIONS_PROMOTION_URL,
+			],
+			'advanced' => [
+				'text' => esc_html__( 'Upgrade Now', 'elementor' ),
+				'url' => self::PRO_TO_ADVANCED_PERMISSIONS_PROMOTION_URL,
+			],
+		];
+
+		$filtered_data = apply_filters( 'elementor/element_manager/admin_app_data/promotion_data/manager_permissions', $promotion_data );
+
+		foreach ( $filtered_data as $key => $data ) {
+			$filtered_data[ $key ] = $this->validate_promotion_data( $filtered_data[ $key ], $promotion_data[ $key ] );
+		}
+
+		return array_replace( $promotion_data, $filtered_data );
+	}
+
+	private function get_element_manager_promotion(): array {
+		$promotion_data = [
+			'text' => esc_html__( 'Upgrade Now', 'elementor' ),
+			'url' => self::ELEMENT_MANAGER_PROMOTION_URL,
+		];
+
+		$filtered_data = apply_filters( 'elementor/element_manager/admin_app_data/promotion_data/element_manager', $promotion_data );
+
+		$this->validate_promotion_data( $filtered_data, $promotion_data );
+
+		return array_replace( $promotion_data, $filtered_data );
+	}
+
+	private function validate_promotion_data( $filtered_data, $promotion_data ) {
+		if ( Validate_Promotion::domain_is_on_elementor_dot_com( $filtered_data['url'] ) ) {
+			$filtered_data['url'] = esc_url( $filtered_data['url'] );
+		}
```

Comment:
@davseve Not sure if this part makes sense.

You are returning `$filtered_data`, so if `domain_is_on_elementor_dot_com()` dis not pass, it will just return the malicious URL.

I would expect something like:

```
if ( ! Validate_Promotion::domain_is_on_elementor_dot_com( $filtered_data['url'] ) ) {
		unset( $filtered_data['url'] );
}
```

This way it won't be replaced in `array_replace` and keep the original URL.

Reply:
Good comment, I missed that.
Fixed.